File: src/comun/maquina/Maquina.java
package comun.maquina;

import comun.componente.*;
import java.util.List;

/**
 *
 * @author Barrionuevo Diego
 */
public class Maquina {

    private String id;//int?
    private String nombre;
    private String fabricante;
    private String placeMadre;//deberia ser una clase?
    private Bios bios;
    private SistemaOperativo sistemaOperativo;
    private DiscoDuro discoDuro;//podria ser un array
    private List<MemoriaRam> memoriasRam;
    private Procesador procesador;

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public String getPlaceMadre() {
        return placeMadre;
    }

    public void setPlaceMadre(String placeMadre) {
        this.placeMadre = placeMadre;
    }

    public Bios getBios() {
        return bios;
    }

    public void setBios(Bios bios) {
        this.bios = bios;
    }

    public SistemaOperativo getSistemaOperativo() {
        return sistemaOperativo;
    }

    public void setSistemaOperativo(SistemaOperativo sistemaOperativo) {
        this.sistemaOperativo = sistemaOperativo;
    }

    public DiscoDuro getDiscoDuro() {
        return discoDuro;
    }

    public void setDiscoDuro(DiscoDuro discoDuro) {
        this.discoDuro = discoDuro;
    }

    public void agregarMemoriasRam(MemoriaRam memoriaRam) {
        this.memoriasRam.add(memoriaRam);
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public List<MemoriaRam> getMemoriasRam() {
        return memoriasRam;
    }

    public void setMemoriasRam(List<MemoriaRam> memoriasRam) {
        this.memoriasRam = memoriasRam;
    }

    public Procesador getProcesador() {
        return procesador;
    }

    public void setProcesador(Procesador procesador) {
        this.procesador = procesador;
    }

    public Maquina() {
    }

}

File: Cliente/Linux/componentes/maquina.py
class Maquina():    
    
    def __init__(self):
        self.ipv4 = ""        
        self.ipv6 = ""
        self.mac = ""
        self.nombre = ""
        self.fabricante = ""
        self.sistema_operativo = ""
        self.memorias_ram = []       
        self.discos_duro = []       
        
    def setid(self, id):
        self.id = id
    
    def setipv4(self, ipv4):
        self.ipv4 = ipv4
    
    def setipv6(self, ipv6):
        self.ipv4 = ipv6
    
    def setmac(self, mac):
        self.ipv4 = mac

    def setnombre(self, nombre):
        self.nombre = nombre
    
    def setfabricante(self, fabricante):
        self.fabricante = fabricante

    def setplaca_madre(self, placa_madre):
        self.placa_madre = placa_madre

    def setbios(self, bios):
        self.bios = bios
    
    def setsistemaoperativo(self, sistema_operativo):
        self.sistema_operativo = sistema_operativo
    
    def setdiscosduro(self, discos_duro):
        self.discos_duro = discos_duro
    
    def setmemoriasram(self, memorias_ram):
        self.memorias_ram = memorias_ram
    
    def setprocesador(self, procesador):
        self.procesador = procesador
    
    def tostr(self):
        maquina = 'Nombre:' + self.nombre + '\n' + 'Fabricante:' + self.fabricante + '\n' + 'SistemaOperativo:' + self.sistema_operativo + '\n\n' + 'DiscosDuro:' + self.listardiscosduro() + '\n\n' + 'Procesador:' + self.procesador.tostr() + '\n\n' + 'MemoriasRam:' + self.listarmemoriasram()
        return maquina
    
    def listarmemoriasram(self):
        strmemorias = ""
        for memoria in self.memorias_ram:
            strmemorias = strmemorias + '\n' + memoria.tostr()
        return strmemorias
    
    def listardiscosduro(self):
        strdiscos = ""
        for disco in self.discos_duro:
            strdiscos = strdiscos + '\n' + disco.tostr()
        return strdiscos